pkg/rke2/server.go

```diff
@@ -43,9 +44,14 @@ func Agent(ctx *cli.Context, cfg Config) error {
 }
 
 func setup(ctx *cli.Context, cfg Config) error {
-	dataDir := cmds.ServerConfig.DataDir
-	if dataDir == "" {
-		dataDir = cmds.AgentConfig.DataDir
+	var dataDir string
+	for _, f := range ctx.Command.Flags {
+		switch t := f.(type) {
+		case *cli.StringFlag:
+			if strings.Contains(t.Name, "data-dir") {
```

Comment:
Why is this contains vs equals? Might this cause problems down the road if we add another data-dir related option in the future?

Reply:
It contains a ",d" so ctains cuts to the chase.

Reply:
I haven't dug through the cli library enough to know off the top of my head how the Name for a flag gets built so that feels a little odd but OK.

Reply:
Agreed.